.github/workflows/review-code.yaml

```diff
@@ -0,0 +1,61 @@
+name: Review PR Diff
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - edited
+
+jobs:
+  reviewmycode:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Step 1: Checkout the PR code
+      - name: Checkout PR Code
+        uses: actions/checkout@v2
+
+      # Step 2: Fetch all history for diff
+      - name: Fetch All History for Diff
+        run: git fetch --all
+
+      # Step 3: Get PR Diff with Context and File Paths
+      - name: Get PR Diff and File Paths with Context
+        id: diff
+        run: |
+          echo "Getting PR diff with 3 lines of context"
+          git diff -U3 origin/${{ github.event.pull_request.base.ref }}...origin/${{ github.event.pull_request.head.ref }} > diff.txt
+          git diff --name-only origin/${{ github.event.pull_request.base.ref }}...origin/${{ github.event.pull_request.head.ref }} > file_paths.txt
+
+      # Step 4: Echo Diff and File Paths
+      - name: Echo Diff and File Paths
+        run: |
+          echo "Diff content:"
+          cat diff.txt  # Output the content of diff.txt
+          echo "File paths:"
+          cat file_paths.txt  # Output the content of file_paths.txt
+
+      # Step 5: Send PR Diff for Each File Separately
+      - name: Send PR Diff for Each File Separately
+        run: |
+          diff_content=$(cat diff.txt | jq -Rsa .)
+          # Loop through each file path and send it in a separate request
+          while IFS= read -r file; do
+            echo "Processing file: $file"
+            file_content=$(git show "${{ github.event.pull_request.head.sha }}:$file" | jq -Rsa .)  # Get the full file content
+
+            # Send the diff and file content to the backend for each file separately
+            curl -X POST "https://app5-latest.onrender.com/openai/api/v2/strategy/4/review-pr-diff" \
```

Comment:
Potential security vulnerability: The curl command sends data to an external API without proper input validation or sanitization, which can lead to security risks like injection attacks. (Severity: High)